src/sgame/sg_votes.cpp

```diff
@@ -0,0 +1,1198 @@
+{
+	if ( s.empty() )
+	{
+		return "";
+	}
+	std::string out;
+	out.reserve( s.size() * 2 );
+	for ( const auto& c : s )
+	{
+		out += '\\';
+		out += c;
+	}
+	return out;
+}
+
+// We only want to escape for commands that are executed. For display only, there is no injection
+// concern, so don't escape those.
+static std::string G_HandleVoteTemplate( Str::StringRef str, gentity_t* ent, team_t team,
+                                         std::string& cmd, std::string& arg, std::string& reason,
+                                         std::string& name, int clientNum, int id, bool escape )
+{
+	std::unordered_map<std::string, std::string> params = {
+		{"team",       BG_TeamNamePlural( team )  },
+		{ "arg",       arg						},
+		{ "reason",    reason                     },
+		{ "name",      name					   },
+		{ "slot",      std::to_string( clientNum )},
+		{ "namelogId", std::to_string( id )       },
+		{ "caller",    ent->client->pers.netname  },
+	};
+	std::string out;
+	out.reserve( str.size() + arg.size() + reason.size() );
+	size_t c = 0;
+	size_t s = 0;
+	size_t e = 0;
+	while ( c < str.size() )
+	{
+		s = str.find( MARKER, c );
+		if ( s != std::string::npos && !IsEscapedMarker( str, s ) )
+		{
+			e = str.find( MARKER, s + 1 );
+			if ( e != std::string::npos )
+			{
+				out += str.substr( c, s - c );
+				const auto& param = params[ str.substr( s + 1, e - s - 1 ) ];
+				out += Quote( Cmd::Escape( escape ? VoteEscape( param ) : param ) );
+				c = e + 1;
+			}
+			else
+			{
+				Log::Warn( "Unterminated %c in str: %s", MARKER, str );
+				return "";
+			}
+		}
+		else
+		{
+			out += str.substr( c );
+			c = str.size();
+		}
+	}
+	return out;
+}
+
+bool G_AddCustomVote( std::string vote, VoteDefinition def, std::string voteTemplate,
+                      std::string displayTemplate )
+{
+	if ( vote.empty() )
+	{
+		return false;
+	}
+	auto it = voteInfo.find( vote );
+	if ( it != voteInfo.end() )
+	{
+		return false;
+	}
+	it = voteInfo.emplace( std::move( vote ), std::move( def ) ).first;
+	auto handler = [ vt = move( voteTemplate ), dt = move( displayTemplate ) ](
+					   gentity_t* ent, team_t team, std::string& cmd, std::string& arg,
+					   std::string& reason, std::string& name, int clientNum, int id )
+	{
+		std::string voteCmd =
+			G_HandleVoteTemplate( vt, ent, team, cmd, arg, reason, name, clientNum, id, true );
+		Q_strncpyz( level.team[ team ].voteString, voteCmd.c_str(),
+		            sizeof( level.team[ team ].voteString ) );
+		std::string display =
+			G_HandleVoteTemplate( dt, ent, team, cmd, arg, reason, name, clientNum, id, false );
+		Q_strncpyz( level.team[ team ].voteDisplayString, display.c_str(),
+		            sizeof( level.team[ team ].voteDisplayString ) );
+		return true;
+	};
+	it->second.handler = std::move( handler );
+	return true;
+}
+
+bool G_ParseVoteType( Str::StringRef s, VoteType* type )
+{
+	static const std::unordered_map<std::string, VoteType, Str::IHash> map = {
```

Comment:
You have to put Str::IEqual also to make it actually case insensitive. This applies to the other 3 cases also

Reply:
oops thanks